### DISCUSSION TITLE
Question: How to start generation automatically when redirected to a chat we create via api?

### DISCUSSION THREAD

**Discussion by MarlNox** [open-webui/open-webui#3373] [General]
Hi there, I have a couple of questions, Specifically: 
How to start generation automatically when redirected to a chat we create via api?
Or alternatively how to start the generation via api for a given message we create via the api?

**Comment by Bearium-BB** [comment]
Hi there, I’m not sure if anyone has figured out a solution yet, but I’m using this endpoint to generate the chat: http://192.168.xxx.xxx:8080/api/v1/chats/new. However, I cannot find a way to make the chatbot generate a response. Any help would be greatly appreciated.

**Reply by m8dhouse** [reply]
afaik the only way for now is to use the http parameters (so not via API)
<url>/?q="your query"&models=gpt-4o-mini

**Reply by Bearium-BB** [reply]
Thank you for responding. Do you know if it is possible to include a file in there somehow?

**Reply by m8dhouse** [reply]
not that I know. 
check this though: https://github.com/open-webui/open-webui/discussions/1349#discussioncomment-10456173

**Reply by MarlNox** [reply]
> Hi there, I’m not sure if anyone has figured out a solution yet, but I’m using this endpoint to generate the chat: http://192.168.xxx.xxx:8080/api/v1/chats/new. However, I cannot find a way to make the chatbot generate a response. Any help would be greatly appreciated.

Hey I did figure it out. 
It was a less than elegant solution, but it works. 
In my case I had to redirect the user and generate the chat remotely aswell as the response. 
I did the following changes: 
1. The chat is created with the given Input question, from the user. 
2. When the user loads the pages of that chat for the first time, the regeenrate response is triggered. 
3. I hide the message number only under the first message, and only if the user was redirected (ive included a couple of flags to do the identification) . 
4. When the user loads the pages, the script checks if a response is existing, or if the user was redirected - if a response doesnt exist, and the user has been redirected to the page, then the regenerate is triggered. 
5. From the user's perspective it feels as if the generation started once the page was loaded. 

Hope this helps.
